StyleCop.Analyzers/StyleCop.Analyzers/SpacingRules/SA1024CodeFixProvider.cs

```diff
@@ -0,0 +1,131 @@
+﻿namespace StyleCop.Analyzers.SpacingRules
+{
+    using System.Collections.Immutable;
+    using System.Composition;
+    using System.Threading.Tasks;
+    using Microsoft.CodeAnalysis;
+    using Microsoft.CodeAnalysis.CodeActions;
+    using Microsoft.CodeAnalysis.CodeFixes;
+    using Microsoft.CodeAnalysis.CSharp;
+
+    /// <summary>
+    /// Implements a code fix for <see cref="SA1024ColonsMustBeSpacedCorrectly"/>.
+    /// </summary>
+    /// <remarks>
+    /// <para>To fix a violation of this rule, ensure that the spacing around the colon follows the rule.</para>
+    /// </remarks>
+    [ExportCodeFixProvider(LanguageNames.CSharp, Name = nameof(SA1024CodeFixProvider))]
+    [Shared]
+    public class SA1024CodeFixProvider : CodeFixProvider
+    {
+        private static readonly ImmutableArray<string> FixableDiagnostics =
+            ImmutableArray.Create(SA1024ColonsMustBeSpacedCorrectly.DiagnosticId);
+
+        /// <inheritdoc/>
+        public override ImmutableArray<string> FixableDiagnosticIds => FixableDiagnostics;
+
+        /// <inheritdoc/>
+        public override FixAllProvider GetFixAllProvider()
+        {
+            return WellKnownFixAllProviders.BatchFixer;
+        }
+
+        /// <inheritdoc/>
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+
+            foreach (var diagnostic in context.Diagnostics)
+            {
+                if (!diagnostic.Id.Equals(SA1024ColonsMustBeSpacedCorrectly.DiagnosticId))
+                {
+                    continue;
+                }
+
+                SyntaxToken token = root.FindToken(diagnostic.Location.SourceSpan.Start);
+                if (token.IsMissing)
+                {
+                    continue;
+                }
```

Comment:
This should never occur, if this ever happens, there is a bug in Roslyn. I think this statement should be removed. It would also remove the need to get the syntax root here, improving the performance of this code. The penalty for obtain the syntax root can then be taken when the user activates the code fix.